tpu/src/main/java/tpu/CreateQueuedResource.java

```diff
@@ -0,0 +1,121 @@
+
+    createQueuedResource(
+        projectId, zone, queuedResourceId, nodeName, tpuType, tpuSoftwareVersion);
+  }
+
+  // Creates a Queued Resource
+  public static QueuedResource createQueuedResource(String projectId, String zone,
+      String queuedResourceId, String nodeName, String tpuType, String tpuSoftwareVersion)
+      throws IOException, ExecutionException, InterruptedException {
+    // With these settings the client library handles the Operation's polling mechanism
+    // and prevent CancellationException error
+    TpuSettings.Builder clientSettings =
+        TpuSettings.newBuilder();
+    clientSettings
+        .createQueuedResourceSettings()
+        .setRetrySettings(
+            RetrySettings.newBuilder()
+                .setInitialRetryDelay(Duration.ofMillis(5000L))
+                .setRetryDelayMultiplier(2.0)
+                .setInitialRpcTimeout(Duration.ZERO)
+                .setRpcTimeoutMultiplier(1.0)
+                .setMaxRetryDelay(Duration.ofMillis(45000L))
+                .setTotalTimeout(Duration.ofHours(24L))
+                .build());
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (TpuClient tpuClient = TpuClient.create(clientSettings.build())) {
+      String parent = String.format("projects/%s/locations/%s", projectId, zone);
+      Node node =
+          Node.newBuilder()
+              .setName(nodeName)
+              .setAcceleratorType(tpuType)
+              .setRuntimeVersion(tpuSoftwareVersion)
+              .setQueuedResource(
+                  String.format(
+                      "projects/%s/locations/%s/queuedResources/%s",
+                      projectId, zone, queuedResourceId))
+              .build();
+
+      QueuedResource queuedResource =
+          QueuedResource.newBuilder()
+              .setName(queuedResourceId)
+              .setTpu(
+                  QueuedResource.Tpu.newBuilder()
+                      .addNodeSpec(
+                          QueuedResource.Tpu.NodeSpec.newBuilder()
+                              .setParent(parent)
+                              .setNode(node)
+                              .setNodeId(nodeName)
+                              .build())
+                      .build())
+              // You can request a queued resource using a reservation by specifying it in code
+              //.setReservationName(
+              // "projects/YOUR_PROJECT_ID/locations/YOUR_ZONE/reservations/YOUR_RESERVATION_NAME")
+              .build();
+
+      CreateQueuedResourceRequest request =
+          CreateQueuedResourceRequest.newBuilder()
+              .setParent(parent)
+              .setQueuedResourceId(queuedResourceId)
+              .setQueuedResource(queuedResource)
+              .build();
+
+      // You can wait until TPU Node is READY,
+      // and check its status using getTpuVm() from "tpu_vm_get" sample.
```

Comment:
what is a meaning for this comment? either implement the waiting (mind that you will need to mock this in the tests) or consider showing it in a separate code sample.

Reply:
Thank you! Deleted